vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow/src/main/java/com/vaadin/flow/component/confirmdialog/ConfirmDialog.java

```diff
@@ -179,6 +181,39 @@ public Style getStyle() {
                 "ConfirmDialog does not support adding styles to overlay");
     }
 
+    /**
+     * Sets the "aria-describedby" attribute of the dialog overlay.
+     * <p>
+     * By default, all elements inside the message area are linked through the
+     * `aria-describedby` attribute. However, there are cases where this can
+     * confuse screen reader users (eg. the dialog may present a password
+     * confirmation form). For these cases, it's better to associate only the
+     * elements that will help describe the confirmation dialog through this
+     * API.
+     * <p>
+     * To restore the generated value, pass `null` as argument
+     *
+     * @param describedBy
+     *            the attribute value
+     */
+    public void setAriaDescribedBy(String describedBy) {
+        getElement().setProperty("accessibleDescriptionRef", describedBy);
+    }
+
+    /**
+     * Gets the aria-describedby of the component
+     * <p>
+     * Note that this will only return a value if
+     * {@link #setAriaDescribedBy(String)} was called before.
+     *
+     * @return an optional aria-describedby of the component if no
+     *         aria-describedby has been set
```

Comment:
Something isn't right with this sentence. The method always returns an optional regardless if the property has been set or not.

Reply:
Changed the description. PTAL